### PR TITLE
CVSB-6659

### PR DESCRIPTION
**Pull request by vladddragomir**

### CODE REVIEW

File: src/pages/testing/test-creation/test-cancel/test-cancel.ts [hunk 1]
```diff
@@ -129,11 +129,23 @@ export class TestCancelPage {
           let activity = this.activityService.createActivityBodyForCall(this.visitService.visit, testResult, false);
           this.activityService.submitActivity(activity).subscribe(
             (resp) => {
+              const log: Log = {
+                type: 'info',
```
[the remark this answers is not in the document]
Comment (vladddragomir) [reply src/pages/testing/test-creation/test-cancel/test-cancel.ts]: yeah, i guess i could

[the remark this answers is not in the document]
Comment (vladddragomir) [reply src/pages/testing/test-creation/test-cancel/test-cancel.ts]: done

File: src/pages/testing/test-creation/test-cancel/test-cancel.ts [hunk 2]
```diff
@@ -129,11 +129,23 @@ export class TestCancelPage {
...
             (resp) => {
+              const log: Log = {
+                type: LOG_TYPES.INFO,
+                message: `${this.oid} - ${resp.status} ${resp.statusText} for API call to ${resp.url}`,
+                timestamp: Date.now(),
+              };
+              this.store$.dispatch(new logsActions.SaveLog(log));
               let activityIndex = this.activityService.activities.map((activity) => activity.endTime).indexOf(testResult.testStartTimestamp);
-              if (activityIndex > -1) this.activityService.activities[activityIndex].id = resp.id;
+              if (activityIndex > -1) this.activityService.activities[activityIndex].id = resp.body.id;
```
Comment (agoncearenco) [inline src/pages/testing/test-creation/test-cancel/test-cancel.ts]: Was this a bug?

File: src/providers/global/http.service.ts [hunk 3]
```diff
@@ -54,12 +54,12 @@ export class HTTPService {
     return this.http.put(`${AppConfig.BACKEND_URL_VISIT}/${visitID}/end`, null, {observe: 'response'});
   }
 
-  postActivity(activities: ActivityModel): Observable<any> {
-    return this.http.post(AppConfig.BACKEND_URL_VISIT, activities);
+  postActivity(activities: ActivityModel): Observable<HttpResponse<any>> {
+    return this.http.post(AppConfig.BACKEND_URL_VISIT, activities, {observe: 'response'});
```
Comment (agoncearenco) [inline src/providers/global/http.service.ts]: Did you update all the calls to this service with resp.body.id? Even the ones that were using this method previously?

**Pull request merged**